implant/sliver/registry/registry_windows.go

```diff
@@ -182,3 +188,79 @@ func CreateSubKey(hostname string, hive string, path string, keyName string) err
 	_, _, err = registry.CreateKey(*k, keyName, registry.ALL_ACCESS)
 	return err
 }
+
+func generateTempFileName() string {
+	const charset = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
+	randomBytes := make([]byte, 8)
+	for i := range randomBytes {
+		randomBytes[i] = charset[rand.Intn(len(charset))]
+	}
+	return string(randomBytes) + ".tmp"
+}
+
+// ReadHive dumps the content of a registry hive into a single binary blob
+func ReadHive(hostname string, requestedRootHive string, requestedHive string) ([]byte, error) {
+	// A quick sanity check on the requested root hive
+	/*rootHive, found := hives[requestedRootHive]
+	if !found {
+		return nil, fmt.Errorf("could not find root hive %s", requestedRootHive)
+	}*/
+
+	// In order to dump a hive, we need the SeBackupPrivilege privilege
+	err := priv.SePrivEnable("SeBackupPrivilege")
+	if err != nil {
+		// {{if .Config.Debug}}
+		log.Println("SePrivEnable failed:", err)
+		// {{end}}
+		return nil, err
+	}
+
+	/*
+		Create a random file name to output the hive to.
+		The system call to dump registry information (RegSaveKeyW)
+		only accepts a file name as an output location, as opposed to a
+		buffer. So we will output the result of the system call, read the
+		resulting file into memory, then delete the file.
+	*/
+	tempFileName := fmt.Sprintf("%s\\%s", os.TempDir(), generateTempFileName())
```

Comment:
Same thing here, just call `f, err := os.CreateTemp("", "*")` instead, and retrieve the filename with `f.Name()`.

Reply:
@lesnuages: So funny thing. In my testing, if the file with the file name you pass to the `RegSaveKeyW` API call exists, the file does not get written. Since `os.CreateTemp` creates a file, I had to come up with something to generate a file name.

Reply:
Ack, I didn't realize `RegSaveKeyW` was calling `CreateFile` itself.